kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -444,19 +525,59 @@ class KinematicsBase
                                    std::string* error_text_out = NULL) const;
 
   /**
-   * @brief  Set the search discretization
+   * @brief  Set the search discretization value for all the redundant joints
    */
   void setSearchDiscretization(double sd)
   {
-    search_discretization_ = sd;
+    redundant_joint_discretization_.clear();
+    for(std::vector<unsigned int>::iterator i = redundant_joint_indices_.begin();
+       i != redundant_joint_indices_.end();i++)
+    {
+      redundant_joint_discretization_[*i] = sd;
+    }
+  }
+
+  /**
+   * @brief Sets individual discretization values for each redundant joint.
+   *
+   * Calling this method replaces previous discretization settings.
+   *
+   * @param discretization a map of joint indices and discretization value pairs.
+   */
+  void setSearchDiscretization(const std::map<int,double>& discretization)
+  {
+    redundant_joint_discretization_.clear();
+    redundant_joint_indices_.clear();
+    for(std::map<int,double>::const_iterator i = discretization.begin();
+        i != discretization.end() ; i++)
+    {
+      redundant_joint_discretization_.insert(*i);
+      redundant_joint_indices_.push_back(i->first);
+    }
   }
 
   /**
    * @brief  Get the value of the search discretization
    */
-  double getSearchDiscretization() const
+  double getSearchDiscretization(int joint_index = 0) const
```

Comment:
This might not be such an easy change. Replacing this method with something that returns a map breaks some code in moveit_ros (see below)
```
src/moveit_ros/planning/planning_components_tools/src/kinematics_speed_and_validity_evaluator.cpp:72:75: note:   ‘const std::map<int, double>’ is not derived from ‘const std::basic_string<_CharT, _Traits, _Alloc>’
         ROS_INFO_STREAM("Search res: " << solver->getSearchDiscretization())
```
An extra method that returns a const reference to the map could be added but that feels slightly redundant.